test/FunctionalTests/Microsoft.OData.Core.Tests/JsonLight/ODataJsonLightReaderUtilsTests.cs

```diff
@@ -0,0 +1,90 @@
+        }
+
+        [Fact]
+        public void TesConvertValueNull()
+        {
+            object nullValue = typeReaderUtils.GetMethod("ConvertValue", bindingFlags)
+                .Invoke(null, /*static method*/
+                    new object[] {
+                        null,
+                        EdmCoreModel.Instance.GetInt64(false),
+                        new ODataMessageReaderSettings(), /*ODataMessageReaderSettings*/
+                        false, /*validateNullValue*/
+                        "custome.nullablePropertyName",
+                        null /*ODataPayloadValueConverter*/
+                    }
+                );
+            nullValue.Should().Be(null);
+        }
+
+        [Fact]
+        public void TestGetPayloadTypeName()
+        {
+            Dictionary<object[], string> payloadTypeToNames = new Dictionary<object[], string>();
+            payloadTypeToNames.Add(new object[] {null}, null);
+            payloadTypeToNames.Add(new object[] {true}, OData.Metadata.EdmConstants.EdmBooleanTypeName);
+            payloadTypeToNames.Add(new object[] { "myPayloadItem" }, OData.Metadata.EdmConstants.EdmStringTypeName);
+            payloadTypeToNames.Add(new object[] { Int32.MinValue }, OData.Metadata.EdmConstants.EdmInt32TypeName);
+            payloadTypeToNames.Add(new object[] { Double.Epsilon }, OData.Metadata.EdmConstants.EdmDoubleTypeName);
+
+            ODataCollectionValue collectionValue = new ODataCollectionValue();
+            payloadTypeToNames.Add(new object[] { collectionValue }, EdmLibraryExtensions.GetCollectionTypeFullName(collectionValue.TypeName));
+
+            ODataResource resourceValue = new ODataResource();
+            payloadTypeToNames.Add(new object[] { resourceValue }, resourceValue.TypeName);
+
+            foreach (KeyValuePair<object[], string> pair in payloadTypeToNames)
+            {
+                object result = typeReaderUtils.GetMethod("GetPayloadTypeName", bindingFlags)
+                    .Invoke(null, /*static method*/
+                        pair.Key);
+                result.Should().Be(pair.Value);
+            }
+        }
+
+        [Fact]
+        public void TestGetPayloadTypeNameError()
+        {
+            try
+            {
+                typeReaderUtils.GetMethod("GetPayloadTypeName", bindingFlags)
+                    .Invoke(null, /*static method*/
+                        new object[] {Guid.NewGuid()});
+            }
+            catch (TargetInvocationException e)
+            {
+                e.InnerException.Should().BeOfType<ODataException>();
+                e.InnerException.Message.Should()
+                    .Be("An internal error 'ODataJsonLightReader_ReadResourceStart' occurred.");
+                return;
+            }
+
+            false.Should().BeTrue("Exception is expected to be thrown and test execution should not reach here.");

```

Comment:
@robward-ms The actions.ShouldThrow<ExceptionType>.WithMessage() is not entirely applicable here because it is a Reflection call, which always throws the TargetInvocationException w/ inner exception (here as ODataException). In other words, the ExcpetionType would be always TargetInvocationException but we want to assert the inner exception of ODataException. Therefor e the catch / innerException is used above.
I don't like the false.should().BeTrue() either. But, for FluenctAssertion, I couldn't anything such as Assert.Failed(msg) available in other assertion libraries. I am now updating it to "exceptionThrown.Should().BeTrue()" hopefully it is better.

@AlanWong-MS yes it was compilable.